python/lsst/afw/image/testUtils.py

```diff
@@ -162,12 +150,115 @@ def maskedImagesDiffer(maskedImageArrSet1, maskedImageArrSet2,
             continue
 
         if planeName == "mask":
-            errStr = masksDiffer(maskedImageArrSet1[ind], maskedImageArrSet2[ind], skipMaskArr=skipMaskArr)
+            errStr = imagesDiffer(maskedImageArrSet0[ind], maskedImageArrSet1[ind], skipMask=skipMask,
+                rtol=0, atol=0)
             if errStr:
-                retStrs.append(errStr)
+                errStrList.append(errStr)
         else:
-            errStr = imagesDiffer(maskedImageArrSet1[ind], maskedImageArrSet2[ind],
-                skipMaskArr=skipMaskArr, rtol=rtol, atol=atol)
+            errStr = imagesDiffer(maskedImageArrSet0[ind], maskedImageArrSet1[ind],
+                skipMask=skipMask, rtol=rtol, atol=atol)
             if errStr:
-                retStrs.append("%s planes differ: %s" % (planeName, errStr))
-    return " | ".join(retStrs)
+                errStrList.append("%s planes differ: %s" % (planeName, errStr))
+
+    if errStrList:
+        testCase.fail("%s: %s" % (msg, "; ".join(errStrList)))
+
+def imagesDiffer(image0, image1, skipMask=None, rtol=1.0e-05, atol=1e-08):
+    """!Compare the pixels of two image or mask arrays; return True if close, False otherwise
+
+    @param[in] testCase  unittest.TestCase instance the test is part of;
+                        an object supporting one method: fail(self, msgStr)
+    @param[in] image0  image 0 (an lsst.afw.image.Image, lsst.afw.image.Mask, or transposed numpy array)
+    @param[in] image1  image 1 (an lsst.afw.image.Image, lsst.afw.image.Mask, or transposed numpy array)
+    @param[in] skipMask  mask of pixels to skip, or None to compare all pixels;
+        an afwImage.Mask **or** a numpy array (transposed relative to Mask);
+        all non-zero pixels are skipped
+    @param[in] rtol  maximum allowed relative tolerance; more info below
+    @param[in] atol  maximum allowed absolute tolerance; more info below
+
+    The images are nearly equal if all pixels obey:
+        |val1 - val0| <= rtol*|val1| + atol
+    or, for float types, if nan/inf/-inf pixels match.
+
+    @warning the comparison equation is not symmetric, so in rare cases the assertion
+    may give different results depending on which image comes first.
+
+    @return a string which is non-empty if the images differ
+    """
+    errStrList = []
+    imageArr0   = image0.getArray()   if hasattr(image0,   "getArray") else image0
+    imageArr1   = image1.getArray()   if hasattr(image1,   "getArray") else image1
+    skipMaskArr = skipMask.getArray() if hasattr(skipMask, "getArray") else skipMask
+
+    # sanity check; this is useful because the error message from np.allclose
+    # is opaque for object arrays
+    try:
```

Comment:
I am creating new variables and I don't know a really clear way to do that programmatically. I fear the result would be less obvious than what I have now. But I am open to suggestions.